packages/workbox-core/src/types.ts

```diff
@@ -48,12 +48,23 @@ export interface RouteMatchCallback {
  * Options passed to a `RouteHandlerCallback` function.
  */
 export interface RouteHandlerCallbackOptions {
-  request: Request | string;
-  url?: URL;
+  event: ExtendableEvent;
+  request: Request;
+  url: URL;
   params?: string[] | MapLikeObject;
+}
+
+/**
+ * Options passed to a `ManualHandlerCallback` function.
+ */
+export interface ManualHandlerCallbackOptions {
+  request: Request | string;
   event?: ExtendableEvent;
```

Comment:
Oh, right. We had talked about how using strategies manually outside of an event handler wasn't a great idea.

Reply:
One follow-up from that is that I need to change a lot of the parameter signatures in the `PrecacheController` class to mark the `ExtendableEvent` as being required. (That's probably a good thing, since using `PrecacheController` outside of an event sounds like a recipe for disaster.)

Reply:
This change also filters through to the `MatchWrapperOptions` and `PutWrapperOptions` interfaces, which now need to mark the `event` as required.

Reply:
Not sure if you want to do this as part of this PR or a separate on, but there are many cases in our code where we've had to make an event optional in a type definition due to us not making it required in handlers (e.g. [here](https://github.com/GoogleChrome/workbox/blob/f6cecf266436b4b199d6a80ece4f09a8515609d0/packages/workbox-strategies/src/StrategyHandler.ts#L41)). I did a in VSCode for `event?:`, and lots of cases come up (and there are probably others where the name "event" isn't used).

Also, there are cases where we're conditionally checking whether an event was passed (e.g. [here](https://github.com/GoogleChrome/workbox/blob/f6cecf266436b4b199d6a80ece4f09a8515609d0/packages/workbox-strategies/src/StrategyHandler.ts#L120-L122)), and it probably makes sense to remove those now.

How do you want to handle these?

Reply:
Okay, I can get those updated in this PR as well.

Reply:
Thanks, I went through and updated the corresponding jsdoc comments for a few that where missed (it was easier than going back and forth).